alcs-api/src/comment/comment.controller.ts

```diff
@@ -0,0 +1,100 @@
+import { Mapper } from '@automapper/core';
+import { InjectMapper } from '@automapper/nestjs';
+import {
+  Body,
+  Controller,
+  Delete,
+  ForbiddenException,
+  Get,
+  NotFoundException,
+  Param,
+  Patch,
+  Post,
+  Req,
+  UseGuards,
+} from '@nestjs/common';
+import { ApiOAuth2 } from '@nestjs/swagger';
+import * as config from 'config';
+import { RoleGuard } from '../common/authorization/role.guard';
+import { UserRoles } from '../common/authorization/roles.decorator';
+import { ANY_AUTH_ROLE } from '../common/enum';
+import { CommentDto, CreateCommentDto, UpdateCommentDto } from './comment.dto';
+import { Comment } from './comment.entity';
+import { CommentService } from './comment.service';
+
+@Controller('comment')
+@ApiOAuth2(config.get<string[]>('KEYCLOAK.SCOPES'))
+@UseGuards(RoleGuard)
+export class CommentController {
+  constructor(
+    private commentService: CommentService,
+    @InjectMapper() private autoMapper: Mapper,
+  ) {}
+
+  @Get('/:fileNumber')
+  @UserRoles(...ANY_AUTH_ROLE)
+  async get(
+    @Param('fileNumber') fileNumber,
+    @Req() req,
+  ): Promise<CommentDto[]> {
+    const comments = await this.commentService.fetchComments(fileNumber);
+
+    return comments.map((comment) => ({
```

Comment:
looks like a spot for automapper

Reply:
I did a quick google but couldn't find a way to pass in an extra object (request in this case) to AutoMapper for it to use during mapping. Is it possible?

Reply:
Done! Used automapper and made a new private function to add isEditable